cmd/gossamer/main.go

```diff
@@ -102,6 +102,9 @@ func main() {
 // configuration, loads the keystore, initializes the node if not initialized,
 // then creates and starts the node and node services
 func gossamerAction(ctx *cli.Context) error {
+	if ctx.IsSet(GenesisFlag.Name) {
+		return fmt.Errorf("genesis flag not allowed with start command")
+	}
```

Comment:
cc @noot I think this solves your concerns about the genesis flag being passed with the root action

Reply:
I could probably move this below the arguments check but otherwise I think this is all we need for now to prevent genesis from being used with the root action

Reply:
you should just remove it from the GlobalFlags in flags.go
create InitFlags, add GenesisFlag there, and then update initCommand Flags in main.go to initFlags

Reply:
I just realized export, init, and account are all set to `AllFlags`, this is wrong. the flags are separated in flags.go for a reason, specifically `AccountFlags` which are only relevant for `gossamer account`.

Reply:
separate issue please, this pull request addresses loading the properties from the genesis file and making sure they match and are being set in the configuration when initializing a node

Reply:
I'm also not sure it is "wrong" but maybe it is being approached from a different angle

Reply:
it's definitely a bug, you should not be allowed to use the genesis flag without `init`.

Reply:
you should be able to use with export as well and it should have no affect if you are using it with other commands so I am still not sure if it is wrong or a bug but we can take a look

Reply:
I also don't think this fixes the issue where you `init` with genesisA.json, then run `gossamer --config configB.json` where configB genesis set to genesisB.json.

Reply:
I am not sure this is related to this issue, can you open a new issue?

Reply:
https://github.com/ChainSafe/gossamer/issues/736